vertical-pod-autoscaler/pkg/recommender/input/history/history_provider.go

```diff
@@ -40,8 +40,17 @@ type PrometheusBasicAuthTransport struct {
 }
 
 // RoundTrip function injects the username and password in the request's basic auth header
-func (t *PrometheusBasicAuthTransport) RoundTrip(req *http.Request) (*http.Response, error) {
-	req.SetBasicAuth(t.Username, t.Password)
+func (c *PrometheusHistoryProviderConfig) RoundTrip(req *http.Request) (*http.Response, error) {
+	if c.PrometheusBasicAuthTransport.Username != "" && c.PrometheusBasicAuthTransport.Password != "" {
```

Comment:
Is this a behavior change? Are there use-cases where one wants to only set a username and no password?

Reply:
Hey @kwiesmueller 
Kinda - Prior to this change round tripper was enabled only in case both username and password were supplied. Now round tripper is enabled any how and the logic was moved into the round tripper func. So in terms of username password no behavior changes - Will be used only if both supplied. But behavior did changed in case non will be provided but still round tripper will be called, Hoping I was able to answer your q